elements/rh-table/rh-table.ts

```diff
@@ -0,0 +1,175 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+
+import styles from './rh-table.css';
+import { RequestSortEvent, RhSortButton } from './rh-sort-button.js';
+
+/**
+ * A table is a container for displaying information. It allows a user to scan, examine, and compare large amounts of data.
+ *
+ * @summary Organizes and displays information from a data set
+ *
+ * @slot               - an HTML table
+ * @slot    summary    - a brief description of the data
+ */
+@customElement('rh-table')
+export class RhTable extends LitElement {
+  static readonly styles = [styles];
+
+  private static getNodeContentForSort(
+    columnIndexToSort: number,
+    node: Element,
+  ) {
+    const content = node.querySelector(`
+      :is(th, td):nth-child(${columnIndexToSort + 1}),
+      tr > :is(th, td):nth-child(${columnIndexToSort + 1})
+    `.trim())?.textContent?.trim()?.toLowerCase() ?? '';
+    return { node, content };
+  }
+
+  private static sortByContent(
+    direction: 'asc' | 'desc',
+    a: { content: string },
+    b: { content: string },
+  ) {
+    if (direction === 'asc') {
+      return (a.content < b.content ? -1 : a.content > b.content ? 1 : 0);
+    } else {
+      return (b.content < a.content ? -1 : b.content > a.content ? 1 : 0);
+    }
+  }
+
+  get #table(): HTMLTableElement | undefined {
+    return this.querySelector('table') as HTMLTableElement | undefined;
+  }
+
+  get #cols(): NodeListOf<HTMLTableColElement> | undefined {
+    return this.querySelectorAll('col') as NodeListOf<HTMLTableColElement> | undefined;
+  }
+
+  get #rows(): NodeListOf<HTMLTableRowElement> | undefined {
+    return this.querySelectorAll('tbody > tr') as NodeListOf<HTMLTableRowElement> | undefined;
+  }
+
+  get #summary(): HTMLElement | undefined {
+    return this.querySelector('[slot="summary"]') as HTMLElement | undefined;
+  }
+
+  #logger = new Logger(this);
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.#init();
+  }
+
+  render() {
+    return html`
+        <slot @pointerleave=${this.#onPointerleave}
+              @pointerover=${this.#onPointerover}
+              @request-sort="${this.#onRequestSort}" 
+              @slotchange=${this.#onSlotChange}></slot>
+        <slot id="summary" name="summary"></slot>
```

Comment:
```suggestion
        <slot @pointerleave="${this.#onPointerleave}"
              @pointerover="${this.#onPointerover}"
              @request-sort="${this.#onRequestSort}" 
              @slotchange="${this.#onSlotChange}"></slot>
        <slot id="summary" name="summary"></slot>
```

help treesitter cope with life